openhtf/conf.py

```diff
@@ -294,334 +353,110 @@ def LoadFromDict(self, dictionary, force_reload=False):
 
     Args:
       dictionary: The dictionary to update.
-      force_reload: True to force a load if the config is already loaded.
-    Returns:
-      True if successful.
     """
-    if not force_reload and self._loaded:
-      return False
-
-    self._state.clear()
-    self._state.update(dictionary)
-    self._loaded = True
-    return True
-
-  @threads.Synchronized
-  def LoadMissingFromDict(self, config_dict):
-    """Update any missing configurations from the given dictionary.
-
-    This is similar to dict.update, except that instead of the given
-    dictionary's values overriding the already set values, this function doesn't
-    override. This is due to the fact that these configs can only be retrieved
-    after we've already loaded the authoritative values.
-
-    Args:
-      config_dict: Dictionary from which to load configuration keys and values.
-
-    Raises:
-      ConfigurationNotLoadedError: Raised when updating empty config values.
-    """
-    # Can't update only missing when it's all missing.
-    if not self._loaded:
-      raise ConfigurationNotLoadedError(
-          'Load configuration before updating missing keys.')
-
-    for key, value in config_dict.items():
-      if key in self._state:
+    for key, value in dictionary.iteritems():
+      # Warn in this case.  We raise if you try to access a config key that
+      # hasn't been declared, but we don't raise here so that you can use
+      # configuration files that are supersets of required configuration for
+      # any particular test station.
+      if key not in self._declarations:
+        self._logger.warning('Ignoring undeclared configuration key: %s', key)
```

Comment:
If you want to make superset-configs okay, I'd drop the spamminess of this by storing ignored keys in a list and printing them all out once afterward